.github/workflows/bitcoin-tests.yml

```diff
@@ -1,205 +1,100 @@
-          - tests::epoch_21::test_pox_missing_five_anchor_blocks
-          - tests::epoch_21::test_sortition_divergence_pre_21
-          - tests::epoch_21::test_v1_unlock_height_with_current_stackers
-          - tests::epoch_21::test_v1_unlock_height_with_delay_and_current_stackers
-          - tests::epoch_21::trait_invocation_cross_epoch
-          - tests::epoch_22::pox_2_unlock_all
-          - tests::epoch_22::disable_pox
-          - tests::epoch_22::test_pox_reorg_one_flap
-          - tests::epoch_23::trait_invocation_behavior
-          - tests::neon_integrations::bad_microblock_pubkey
-          - tests::epoch_24::fix_to_pox_contract
-          - tests::epoch_24::verify_auto_unlock_behavior
+          - tests::neon_integrations::test_problematic_txs_are_not_stored
+          - tests::neon_integrations::use_latest_tip_integration_test
+          - tests::should_succeed_handling_malformed_and_valid_txs
     steps:
-      - name: Checkout the latest code
-        id: git_checkout
-        uses: actions/checkout@v3
-      - name: Download docker image
-        id: download_docker_image
-        uses: actions/download-artifact@v3
-        with:
-          name: integration-image.tar.gz
-      - name: Load docker image
-        id: load_docker_image
-        run: docker load -i integration-image.tar.gz && rm integration-image.tar.gz
-      - name: All integration tests with sampled genesis
-        id: bitcoin_integration_tests
-        timeout-minutes: 30
-        env:
-          DOCKER_BUILDKIT: 1
-          TEST_NAME: ${{ matrix.test-name }}
-        run: docker build -o coverage-output --build-arg test_name=${{ matrix.test-name }} -f ./.github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests .
-      - name: Code Coverage
-        id: code_coverage
-        uses: codecov/codecov-action@v3
-        with:
-          files: ./coverage-output/lcov.info
-          name: ${{ matrix.test-name }}
-          fail_ci_if_error: false
+      ## Setup test environment
+      - name: Setup Test Environment
+        id: setup_tests
+        uses: stacks-network/actions/stacks-core/testenv@main
 
-  # Run atlas integration tests
-  atlas-test:
-    name: Atlas Test
-    runs-on: ubuntu-latest
-    needs:
-      - build-integration-image
-    strategy:
-      fail-fast: false
-      matrix:
-        test-name:
-          - tests::neon_integrations::atlas_integration_test
-          - tests::neon_integrations::atlas_stress_integration_test
-    steps:
-      - name: Checkout the latest code
-        id: git_checkout
-        uses: actions/checkout@v3
-      - name: Download docker image
-        id: download_docker_image
-        uses: actions/download-artifact@v3
+      ## Run test matrix using restored cache of archive file
+      ##   - Test will timeout after env.TEST_TIMEOUT minutes
+      - name: Run Tests
+        id: run_tests
+        timeout-minutes: ${{ fromJSON(env.TEST_TIMEOUT) }}
```

Comment:
Curious, does `${{ env.TEST_TIMEOUT }}` not work here? `TEST_TIMEOUT` [does not appear to be a JSON object](https://github.com/stacks-network/stacks-core/pull/4079/files#diff-0e6368aea5b3b1595c45133e133576c67450c749bd978b7b35b284c94a2746e9R19).

Reply:
correct! it's a quirk of yaml and how github actions works. 
since the env var value is being sent to a composite job, it's sending the value as a string (with `"` wrapped around the int). 
since an int with quotes is treated as a string, the [composite aciton](https://github.com/stacks-network/actions/blob/main/stacks-core/run-tests/action.yml) will fail (i suppose an alternative is to have it defined as an input, but it seems unnecessary to add that in the composite action)

the only way i found to not hardcode that timeout in the composite action was to treat it as a json object. 
i could not tell you where i found the documentation on that either - i found it, tried it and it worked so i moved on 😄

Reply:
Interesting! TIL, thanks!